storage/replica_test.go

```diff
@@ -487,6 +487,20 @@ func TestRangeLeaderLease(t *testing.T) {
 	if held, expired := hasLease(tc.rng, tc.clock.Now()); held || !expired {
 		t.Errorf("expected another replica to have expired lease")
 	}
+
+	// Verify that command returns NotLeaderError when lease is rejected.
+	tc.rng.proposeRaftCommandFn = func(id cmdIDKey, cmd roachpb.RaftCommand) <-chan error {
```

Comment:
The other tests create a new replica, which is apparently easily done.  The new test now does the same, there shouldn't be a risk of race anymore.